patches/gen-hardhat-patches.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
```

Comment:
Maybe add this to the `scripts` folder instead?

Reply:
To be honest I don't think it's robust enough and we should probably come up with a solution that directly checks out the repository at a given commit rather than apply a series of commits; I used that approach to quickly minimize the resulting diff, at first.

The script and the patch are checked out to run the tests and benchmarks and to show which commits are necessary on the Hardhat side; we should probably get rid of them before merging, like we did for #545.